File: backend/src/main/java/backend/demo/model/AnalysisData.java
package backend.demo.model;

import java.util.List;

public class AnalysisData {
    private List<Parameter> parameters;
    private double hi2;
    private double[] autocorrelationalFunction;
    private double[] weightedAverageBalances;

    public List<Parameter> getParameters() {
        return parameters;
    }

    public void setParameters(List<Parameter> parameters) {
        this.parameters = parameters;
    }

    public double getHi2() {
        return hi2;
    }

    public void setHi2(double hi2) {
        this.hi2 = hi2;
    }

    public double[] getAutocorrelationalFunction() {
        return autocorrelationalFunction;
    }

    public void setAutocorrelationalFunction(double[] autocorrelationalFunction) {
        this.autocorrelationalFunction = autocorrelationalFunction;
    }

    public double[] getWeightedAverageBalances() {
        return weightedAverageBalances;
    }

    public void setWeightedAverageBalances(double[] weightedAverageBalances) {
        this.weightedAverageBalances = weightedAverageBalances;
    }
}

File: frontend/src/app/models/analysis-data.ts
import {Parameter} from "./parameter";

export class AnalysisData {
  private _parameters: Parameter[] ;
  private _hi2: number;
  private _autocorrelationalFunction: number[];
  private _weightedAverageBalances: number[]


  get parameters(): Parameter[] {
    return this._parameters;
  }

  set parameters(value: Parameter[]) {
    this._parameters = value;
  }

  get hi2(): number {
    return this._hi2;
  }

  set hi2(value: number) {
    this._hi2 = value;
  }


  get autocorrelationalFunction(): number[] {
    return this._autocorrelationalFunction;
  }

  set autocorrelationalFunction(value: number[]) {
    this._autocorrelationalFunction = value;
  }

  get weightedAverageBalances(): number[] {
    return this._weightedAverageBalances;
  }

  set weightedAverageBalances(value: number[]) {
    this._weightedAverageBalances = value;
  }
}
